tests/test_cellranger.py

```diff
@@ -0,0 +1,77 @@
+import os
+import pytest
+import shutil
+
+from tests.util import s3, hash
+import toolchest_client as toolchest
+
+toolchest_api_key = os.environ.get("TOOLCHEST_API_KEY")
+if toolchest_api_key:
+    toolchest.set_key(toolchest_api_key)
+
+MIN_EXPECTED_ARCHIVE_SIZE = 34000000
+MAX_EXPECTED_ARCHIVE_SIZE = 38000000
+
+EXPECTED_SUMMARY_SIZE = 2744825
+EXPECTED_RAW_MATRIX_SIZE = 868393
+EXPECTED_RAW_MATRIX_HASH = "d00cca1d2b4344b03946eeaeedc17ed5"
+EXPECTED_FILTERED_MATRIX_SIZE = 503956
+
+
+@pytest.mark.integration
+def test_cellranger_count():
+    test_dir = "test_cellranger_count"
+    input_dir_path = f"./{test_dir}/inputs/"
+    output_dir_path = f"./{test_dir}/output/"
+    os.makedirs(input_dir_path, exist_ok=True)
+    os.makedirs(output_dir_path, exist_ok=True)
+
+    # Test using a compressed archive in S3
+    output = toolchest.cellranger_count(
+        inputs="s3://toolchest-integration-tests/cellranger/count/pbmc_1k_v3_fastqs_trimmed.tar.gz",
+        transcriptome_name="GRCh38",
+    )
+    verify_cellranger_count_outputs(output, output_dir_path)
+
+    shutil.rmtree(output_dir_path)
+    os.makedirs(output_dir_path, exist_ok=True)
+
+    # Test from a directory of local inputs
+    packed_inputs_path = f"./{test_dir}/inputs.tar.gz"
+    s3.download_integration_test_input(
+        s3_file_key="cellranger/count/pbmc_1k_v3_fastqs_trimmed.tar.gz",
+        output_file_path=packed_inputs_path,
+    )
+    shutil.unpack_archive(packed_inputs_path, input_dir_path)
+    output = toolchest.cellranger_count(
+        inputs=input_dir_path,
+        transcriptome_name="GRCh38",
+    )
+    verify_cellranger_count_outputs(output, output_dir_path)
+
+
+def verify_cellranger_count_outputs(output, output_dir_path):
```

Comment:
Is there a benefit to this being a separate function from `test_cellranger_count`? Looks like this isn't generalizable (e.g. locked hashes)

Reply:
This is separated into its own function to prevent copy-pasting the same code. For scoping, we can bring this within `test_cellranger_count()`.

Reply:
Ah I missed that it was used twice.

Adding followup comment, then.